### PR TITLE
Feature/social config

### PR DESCRIPTION
**Pull request by kuiche**
#### What does this do?
Moves social handling to CMS

#### How should this be manually tested?
Include 
```
{{
    render(controller('Message:Mothership:CMS::Controller:Module:Social#share', {
        page: page
    }))
}}

{{
    render(controller('Message:Mothership:CMS::Controller:Module:Social#links'))
}}
```
into a view. also, copy the fixture into config and add some data.

Social links should show and link to correct places

### CODE REVIEW

File: src/Controller/Module/Social.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class Social extends Controller
+{
+	public function share(Page $page, $description = null, File $image = null)
+	{
+		$schemeAndHost = $this->get('http.request.master')->getSchemeAndHttpHost();
+		$trimmed       = rtrim($page->slug, '/');
+		$uri           = $schemeAndHost . $this->generateUrl('ms.cms.frontend', ['slug' => !empty($trimmed) ? $trimmed : $page->slug]);
+	
+		return $this->render('Message:Mothership:CMS::modules:social:share', [
+			'networks'    => $this->get('cfg')->social,
```
Comment (thomasjthomasj) [inline src/Controller/Module/Social.php]: Why is this called networks?

File: src/Controller/Module/Social.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+			'title'       => $page->metaTitle ?: $page->title,
+			'description' => $description ?: $page->metaDescription,
+			'imageUri'    => $image ? $schemeAndHost . $image->getUrl() : null,
+		]);
+	}
+
+	public function links()
+	{
+		return $this->render('Message:Mothership:CMS::modules:social:links', [
+			'networks' => $this->get('cfg')->social,
```
Comment (thomasjthomasj) [inline src/Controller/Module/Social.php]: As above

File: src/Page/Edit.php [hunk 3]
```diff
@@ -81,6 +81,7 @@ public function save(Page $page)
 				page.updated_by            = :updatedBy?i,
 				page.meta_title            = :metaTitle?s,
 				page.meta_description      = :metaDescription?s,
+				page.meta_image            = :metaImage?i,
```
Comment (thomasjthomasj) [inline src/Page/Edit.php]: This should be ?in to allow null

File: resources/view/modules/meta.html.twig [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<meta charset="UTF-8">
+<meta property="og:title" content="{{ page.metaTitle|default(page.title ~ ' | ' ~ app.cfg.app.name) }}" />
+<meta property="og:description" content="{{ page.metaDescription|default(page.title ~ ' | ' ~ app.cfg.app.name) }}" />
+<meta name="viewport" content="initial-scale=1.0, width=device-width, height=device-height"/>
+<meta name="description" content="{{ page.metaDescription|default(page.title ~ ' | ' ~ app.cfg.app.name) }}">
+<title>{% block title %}{{ page.metaTitle|default(page.title ~ ' | ' ~ app.cfg.app.name) }}{% endblock %}</title>
+<link rel="icon" type="image/png" href="{{ getResizedUrl(page.metaImage, 500, 'AUTO') }}">
+<meta property="og:image" content="{{ getResizedUrl(page.metaImage, 500, 'AUTO') }}" />
```
Comment (thomasjthomasj) [inline resources/view/modules/meta.html.twig]: Should there perhaps be an if statement around the meta image?

File: resources/migrations/_1427287552_MetaImage.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+{
+	public function up()
+	{
+		$this->run("
+			ALTER TABLE
+				page
+			ADD
+				meta_image INTEGER DEFAULT NULL
+			AFTER
+				`type`
```
Comment (thomasjthomasj) [inline resources/migrations/_1427287552_MetaImage.php]: Should be after `meta_description`, at the moment it comes before any of the other meta columns

**Comment by thomasjthomasj** [comment]
screen shot 2015-03-26 at 16 09 15
Got this error when saving a page from the metadata tab after not adding anything

File: src/Page/Edit.php (hunk 3 above)
Comment (kuiche) [inline src/Page/Edit.php]: maybe, didn't realise that was a thing

**Pull request merged**